src/lib.rs

```diff
@@ -3789,6 +3801,448 @@ pub struct SemanticHighlightingParams {
+}
+
+#[cfg(feature = "proposed")]
+impl From<SemanticTokens> for SemanticTokensEditResult {
+    fn from(from: SemanticTokens) -> Self {
+        SemanticTokensEditResult::Tokens(from)
+    }
+}
+
+#[cfg(feature = "proposed")]
+impl From<SemanticTokensEdit> for SemanticTokensEditResult {
+    fn from(from: SemanticTokensEdit) -> Self {
+        SemanticTokensEditResult::TokensEdits(from)
+    }
+}
+
+#[cfg(feature = "proposed")]
+impl From<SemanticTokensPartialResult> for SemanticTokensEditResult {
+    fn from(from: SemanticTokensPartialResult) -> Self {
+        SemanticTokensEditResult::PartialTokens(from)
+    }
+}
+
+#[cfg(feature = "proposed")]
+impl From<SemanticTokensEditsPartialResult> for SemanticTokensEditResult {
+    fn from(from: SemanticTokensEditsPartialResult) -> Self {
+        SemanticTokensEditResult::PartialTokensEdit(from)
+    }
+}
+
+/**
+ * @since 3.16.0 - Proposed state
+ */
+#[derive(Debug, Eq, PartialEq, Clone, Default, Deserialize, Serialize)]
+#[serde(rename_all = "camelCase")]
+#[cfg(feature = "proposed")]
+pub struct SemanticTokensEdits {
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub result_id: Option<String>,
+    /**
+     * For a detailed description how these edits are structured pls see
+     * https://github.com/microsoft/vscode-extension-samples/blob/5ae1f7787122812dcc84e37427ca90af5ee09f14/semantic-tokens-sample/vscode.proposed.d.ts#L131
+     */
+    pub edits: Vec<SemanticTokensEdit>,
+}
+
+/**
+ * @since 3.16.0 - Proposed state
+ */
+#[derive(Debug, Eq, PartialEq, Clone, Default, Deserialize, Serialize)]
+#[serde(rename_all = "camelCase")]
+#[cfg(feature = "proposed")]
+pub struct SemanticTokensEditsPartialResult {
+    pub edits: Vec<SemanticTokensEdit>,
+}
+
+/**
+* Capabilities specific to the `textDocument/semanticTokens`
+*
+* @since 3.16.0 - Proposed state
+*/
+#[derive(Debug, Eq, PartialEq, Clone, Default, Deserialize, Serialize)]
+#[serde(rename_all = "camelCase")]
+#[cfg(feature = "proposed")]
+pub struct SemanticTokensClientCapabilities {
+    /**
+     * Whether implementation supports dynamic registration. If this is set to `true`
+     * the client supports the new `(TextDocumentRegistrationOptions & StaticRegistrationOptions)`
+     * return value for the corresponding server capability as well.
+     */
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub dynamic_registration: Option<bool>,
+
+    /**
+     * The token types known by the client.
+     */
+    pub token_types: Vec<SemanticTokenType>,
+
+    /**
+     * The token modifiers known by the client.
+     */
+    pub token_modifiers: Vec<SemanticTokenModifier>,
+}
+
+#[derive(Debug, Eq, PartialEq, Clone, Deserialize, Serialize)]
+#[serde(rename_all = "camelCase")]
+#[serde(untagged)]
+#[cfg(feature = "proposed")]
+pub enum SemanticTokensDocumentProvider {
+    Bool(bool),
+
+    /// The server supports deltas for full documents.
+    Edits(Option<bool>),
```

Comment:
I think this should be

```rust
Edits {
  edits: Option<bool>
}
```

Reply:
Ah I knew I could do something like that!